code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,305 @@
+			if(boosted == 1)
+				SEND_SOUND(mychild, sound('sound/effects/magic.ogg'))
+				to_chat(mychild, "<b>Someone has activated your crevice.  You will be returned to fight shortly, get ready!</b>")
+				sleep(40)
+			sleep(30)
+			mychild.forceMove(loc)
+			visible_message("<span class='boldwarning'>[mychild] emerges from the crevice!</span>")
+			playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, 1, 1)
+			mychild.revive(full_heal = 1, admin_revive = 1)
+			if(boosted == 1)
+				mychild.maxHealth = mychild.maxHealth * 2
+				mychild.health = mychild.maxHealth
+			
+			return
+			
+/obj/structure/elite_crevice/Initialize()
+	. = ..()
+	AddComponent(/datum/component/gps, "Menacing Signal")
+		
+/obj/structure/elite_crevice/attackby(obj/item/W, mob/user, params)
+	if(istype(W, /obj/item/organ/regenerative_core) && activity == 0 && boosted == 0)
+		var/obj/item/organ/regenerative_core/core = W
+		if(core.preserved == 1)
+			visible_message("<span class='boldwarning'>As [user] drops the core into the crevice, the red light intensifies for a brief moment, then returns to normal.</span>")
+			boosted = 1
+			qdel(core)
+			
+/obj/structure/elite_crevice/proc/arena_trap()
+	var/turf/T = get_turf(src)
+	if(T && activity == 1)
+		for(var/t in RANGE_TURFS(12, T))
+			if(t && get_dist(t, T) == 12)
+				var/obj/effect/temp_visual/elite_crevice_wall/newwall
+				newwall = new /obj/effect/temp_visual/elite_crevice_wall(t, src)
+				newwall.activator = src.activator
+				newwall.ourelite = src.mychild
+		sleep(100)
+		if(src) //Checking to see if we still exist
+			INVOKE_ASYNC(src, .proc/arena_trap)  //Gets another arena trap queued up for when this one runs out.
+			INVOKE_ASYNC(src, .proc/border_check)  //Checks to see if our fighters got out of the arena somehow.
+		return
+
+/obj/effect/temp_visual/elite_crevice_wall
+	name = "magic wall"
+	icon = 'icons/turf/walls/hierophant_wall_temp.dmi'
+	icon_state = "wall"
+
+	duration = 100
+	smooth = SMOOTH_TRUE
+	layer = BELOW_MOB_LAYER
+	var/mob/living/carbon/human/activator = null
+	var/mob/living/simple_animal/hostile/asteroid/elite/ourelite = null
+	color = rgb(255,0,0)
+	light_range = MINIMUM_USEFUL_LIGHT_RANGE
+	light_color = LIGHT_COLOR_RED
+	
+/obj/effect/temp_visual/elite_crevice_wall/Initialize(mapload, new_caster)
+	. = ..()
+	queue_smooth_neighbors(src)
+	queue_smooth(src)
+
+/obj/effect/temp_visual/elite_crevice_wall/Destroy()
+	queue_smooth_neighbors(src)
+	return ..()
+
+/obj/effect/temp_visual/elite_crevice_wall/CanPass(atom/movable/mover, turf/target)
+	if(mover == ourelite || mover == activator)
+		return FALSE
+	else
+		return TRUE
+		
+/obj/structure/elite_crevice/proc/border_check()
+	if(activity == 1)
+		if(activator != null && get_dist(src, activator) >= 12)
+			if(loc)
+				activator.forceMove(loc)
+				visible_message("<span class='boldwarning'>[activator] suddenly reappears above the crevice!</span>")
+				playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, 1, 1)
```

Comment:
Inb4 actionninja complains about booleans in playsound.

Reply:
Sure.